src/Functions/in.cpp

```diff
@@ -125,7 +125,7 @@ class FunctionIn : public IFunction
             const DataTypes & tuple_types = type_tuple->getElements();
             size_t tuple_size = tuple_columns.size();
             for (size_t i = 0; i < tuple_size; ++i)
-                columns_of_key_columns.insert({ tuple_columns[i], tuple_types[i], "" });
+                columns_of_key_columns.insert({ tuple_columns[i], tuple_types[i], tuple->getColumnName(i) });
```

Comment:
Do we really need to store names of tuple elements in column? Looks like we always can get them from type.